j2v/utils/helpers.py

```diff
@@ -53,13 +53,24 @@ def is_ISO_Time(dim_val):
 
 def is_unix_timestamp(dim_val):
     """
-       Checks if dim_val represents a timestamp
-       :param dim_val:
-       :return: True only if string represents a timestamp
-       """
-    date_now = datetime.now()
-    time_diff = timedelta(days=365)
-    return dim_val > 0 and date_now + time_diff > datetime.fromtimestamp(dim_val) > date_now - time_diff
+    Checks if dim_val represents a timestamp if in seconds, milliseconds and microseconds
+    :param dim_val:
+    :return: True only if string represents a timestamp
+    """
+    try:
+        timestamp_now = datetime.timestamp(datetime.now())
+        timestamp_len = len(str(dim_val))
+        if dim_val > 0:
+            if timestamp_len == 13:
+                is_unix_timestamp(dim_val/10**3)
+            elif timestamp_len == 16:
+                is_unix_timestamp(dim_val/10**6)
+            elif timestamp_len == 10:
+                map(datetime.fromtimestamp, [dim_val, timestamp_now - 3.154e+7, timestamp_now + 3.154e+7])
```

Comment:
1. This check will be never executed. To understand why, try the following in your python console:
`def foo(x):
    raise Exception`
`foo(1)`
Then do what you are trying to achieve here:
`map(foo,[1,2,3,4])`
What's the learning?
2. I don't get this numbers `timestamp_now - 3.154e+7, timestamp_now + 3.154e+7`, what's the intent of checking today-1y or today+1y?